crates/iroha_core/src/sumeragi/main_loop.rs

```diff
@@ -841,57 +838,74 @@ impl Sumeragi {
             .expect("INTERNAL BUG: transactions in block exceed usize::MAX");
         let block_time = state.world.view().parameters.sumeragi.block_time();
         let tx_cache_full = self.transaction_cache.len() >= max_transactions.get();
-        let deadline_reached = self.round_start_time.elapsed() > block_time;
+        let deadline_reached =
+            self.topology.view_change_index() > 0 || self.round_start_time.elapsed() > block_time;
```

Comment:
```suggestion
        let deadline_reached =
            self.topology.view_change_index() > 0 || self.round_start_time.elapsed() > block_time;
```

this is abuse of the word `deadline`. view change index is not related to deadline so keep it out